src/components/Input/index.tsx

```diff
@@ -0,0 +1,121 @@
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const userInput = e.currentTarget.value;
+    const filteredList = suggestions.filter(
+      (suggestion) =>
+        suggestion.toLowerCase().indexOf(userInput.toLowerCase()) > -1
+    );
+
+    setSearchTerm(e.currentTarget.value);
+    setFilteredSuggestions(filteredList);
+    setActiveSuggestionIndex(0);
+    setShowSuggestions(true);
+    setShowClearButton(true);
+  };
+
+  const clearInput = () => {
+    setSearchTerm("");
+    setShowClearButton(false);
+  };
+
+  const chooseSuggestion = (e: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
+    e.preventDefault();
+    setFilteredSuggestions([]);
+    setSearchTerm(e.currentTarget.innerText);
+    setActiveSuggestionIndex(0);
+    setShowSuggestions(false);
+  };
+
+  //It's working, but I would like to prevent behaviour that
+  //after I hit the last suggestion the background style don't vanish
+
+  const onKeyDown = (e: KeyboardEvent) => {
+    if (e.key === "Enter") {
+      if (activeSuggestionIndex !== -1) {
+        e.preventDefault();
+        setSearchTerm(filteredSuggestions[activeSuggestionIndex]);
+        setFilteredSuggestions([]);
+        setActiveSuggestionIndex(0);
+        setShowSuggestions(false);
+      }
+    } else if (e.key === "ArrowUp") {
+      if (activeSuggestionIndex >= 0) {
+        e.preventDefault();
+        setActiveSuggestionIndex(activeSuggestionIndex - 1);
+      }
+    } else if (e.key === "ArrowDown") {
+      if (activeSuggestionIndex < filteredSuggestions.length) {
+        e.preventDefault();
+        setActiveSuggestionIndex(activeSuggestionIndex + 1);
+      }
+    }
+  };
+
+  return (
+    <FormWrapper>
+      <Form
+        action="/"
+        method="get"
+        onSubmit={(e) => {
+          e.preventDefault();
+        }}
+      >
+        <Label htmlFor="header-search">
+          <Input
+            type="text"
+            placeholder="Szukaj problemów matematycznych"
+            aria-label="search-input"
+            aria-owns="autocomplete-options"
+            aria-autocomplete="list"
+            value={searchTerm}
+            onChange={handleChange}
+            onKeyDown={(e) => onKeyDown(e)}
+            onSubmit={(e) => {
+              e.preventDefault();
+            }}
+          />
+          {showSuggestions && searchTerm && (
+            <SuggestionsListComponent
+              filteredSuggestions={filteredSuggestions}
+              activeSuggestionIndex={activeSuggestionIndex}
+              chooseSuggestion={chooseSuggestion}
+            />
+          )}
+          <SearchIcon />
+          {showClearButton && searchTerm && (
+            <Button aria-label="Remove button" onClick={() => clearInput()}>
+              <span id="Remove button" hidden>
+                Remove input text button
+              </span>
+
+              <RemoveIcon aria-hidden="true" focusable="false" />
```

Comment:
Hmm, I think we should remove those properties suggesting the element should not be interactive, as you should be able to interact with it using keyboard. ;)
https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard